vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -22,13 +29,18 @@ WindowsModalHostComponentView::WindowsModalHostComponentView(
     const winrt::Microsoft::ReactNative::Composition::Experimental::ICompositionContext &compContext,
     facebook::react::Tag tag,
     winrt::Microsoft::ReactNative::ReactContext const &reactContext)
-    : Super(
-          WindowsModalHostComponentView::defaultProps(),
-          compContext,
-          tag,
-          reactContext,
-          ComponentViewFeatures::Default & ~ComponentViewFeatures::Background) {
-  m_context = reactContext; // save context
+    : Super(compContext, tag, reactContext) {
+  m_reactContext = reactContext; // save react context
+  m_compositionContext = compContext; // save composition context
+}
+
+WindowsModalHostComponentView::~WindowsModalHostComponentView() {
+  // Check if the window handle (m_hwnd) exists and destroy it if necessary
+  if (m_hwnd) {
+    // Close the modal window
+    HideOnUIThread();
```

Comment:
I think we need to destory the window, not just hide it.  Otherwise we will be leaking windows everytime we show a modal.